File: utils/rust/src/puzzle.rs
use std::fs;
use std::fs::File;
use std::io::{prelude::*, BufReader};
use std::path::Path;

pub struct PuzzleInput {
    data: String,
}

impl PuzzleInput {
    pub fn new(path: impl AsRef<Path>) -> Self {
        let data = fs::read_to_string(path)
            .expect("Unable to read file")
            .trim()
            .to_string();
        PuzzleInput { data }
    }

    pub fn string(&self) -> &str {
        self.data.as_str()
    }
    pub fn lines(&self) -> Vec<&str> {
        self.data.split("\n").collect()
    }
}

File: utils/python/python/puzzle.py
import os


class Puzzle(list):
    def __init__(self, path: str | os.PathLike) -> None:
        with open(path, "r") as f:
            self._data = f.read().strip()

        super().__init__(self._data.split("\n"))

    def __str__(self) -> str:
        return self._data
